CHANGES.md

```diff
@@ -66,6 +66,59 @@ The syntactic sugar for effect and record fields which lets one omit the `()` ha
 
 It is now possible to annotate type variables with `Mono` restriction, e.g. `sig id : (a::(Any,Mono)) -> a::(Any,Mono)`.
 
+### Recursive rows
+
+* We can now enter a recursive row effect type, such as `{ |(mu a.F:(() { |a}-> ()) {}-> b|c)}`.
+* **Breaking change**: recursive rows of different types cannot mix anymore - a variant can only contain fields with variant syntax (i.e. uppercase and separated by `|`) in its recursive row variable, and the same goes for others (e.g. records - lowecase, separated by `,`);
+  * these are correct: `[|(mu a. Foo:()|Bar:()|d)|]`, `(|(mu a. foo:(),bar:()|d))`
+  * these are not: `[|(mu a. foo:()|bar:()|d)|]`, `(|(mu a. foo:()|bar:()|d))`
+* **Breaking change**: Recursive variants with no directly exposed fields no longer require (nor allow) the vertical bar separating fields from the row variable: the following is no longer correct: `[| |(mu a . Foo:()|Bar:()|d)|]` (note the bar before `mu`)
+
+## Roundtrip: New pretty printer for types
+
+This version of Links introduces a new pretty printer for types, called Roundtrip. This fixes various round-tripping issues.
+
+The Roundtrip printer is now active by default. The old printer is still present.
+
+The printer(s) to be used can be selected using the setting `types_pretty_printer_engine`, with the following values:
+
+  * `roundtrip`: the new printer
+  * `old`: the original printer
+  * `derived`: no pretty printing - prints the OCaml representation of the types
+
+Note that one can select multiple printers at once, for comparison; this is done by separating printer names by commas, e.g.:
+
+```links
+@set types_pretty_printer_engine "roundtrip,old";
+```
+
+## Effect Syntactic Sugar
+
+**Breaking change** (for code using effect sugar): This version implements enhanced syntactic sugar for effects. The changes influence both the Roundtrip printer (see above) and the desugaring passes (between parsing and typechecking).
```

Comment:
I don't think this is a breaking change or need to be. The default preset should do whatever `effect_sugar=true` used to do.

Reply:
I believe the breaking part comes from the implicit sharing inference: the rule `final_arrow`. I think this can just be removed from the default and that will restore the original behaviour. 

Shouldn't cause problems with tests either, but I'm not too sure of this.

I'll have a look

Reply:
I've opened #1045 to return the option of the original sharing rule for arrow + alias.

However, I've realised that the changes listed here are likely not breaking on their own - the new version of sharing between last arrow + type alias is a more general approach that handles old cases as they were expected, and also handles new cases. 

However, the problem is that #1041 (incorrect behaviour in the type inference engine/unifier) interacts with this new general approach (which relies on correct type inference to infer whether the last arrow will share its effect var with a following alias) and can break existing code - which I think is why I originally marked this as breaking.

For reference see #1032 and #1041.

Reply:
I've removed the "breaking" mark, since the effect sugar itself should not actually break stuff - it's just more general.

However I think the above is still a problem